runtime/codert_vm/decomp.cpp

```diff
@@ -259,10 +259,11 @@ jitResetAllMethods(J9VMThread *currentThread)
 		U_32 methodCount = clazz->romClass->romMethodCount;
 
 		while (methodCount != 0) {
-			if (0 == ((UDATA)method->extra & J9_STARTPC_NOT_TRANSLATED)) {
+			UDATA extra = (UDATA)method->extra;
```

Comment:
Note, this method is called while holding exclusiveVMAccess so it shouldn't matter whether the `extra` field is reread or not.

Reply:
I don't believe the comp thread holds exclusive vm access when it updates the start PC, so even if this thread holds exclusive VM Access, the compilation thread could still update the start PC when it calls `jitMethodTranslated`.

Reply:
I've been looking through the JIT code and I don't see it acquire / release vm access which is surprising to me.  I wouldn't expect the JIT to get exclusive, but I would have expected "regular" shared vmaccess

Reply:
@gacholio do you know what the contract for `jitMethodTranslated` is - whether it needs to hold vmaccess or not?

Reply:
Found it:
https://github.com/eclipse/openj9/blob/05119f696f85cc2137d451e4b6001b94c929b29a/runtime/compiler/control/CompilationThread.cpp#L10052-L10056

I haven't dug back far enough to see the acquire vm access call but the comment indicates this should only occur under vmaccess

Reply:
Yes, that's right.

Reply:
Looks like there's a comment at the top of `compilationEnd` that states that vmaccess must be acquired. It looks like this happens quite high up, so I guess at this point in the code, we must have vmaccess in hand.

Reply:
Want to put these changes back in?  My comment at the top of the review was:
> Note, that most of these changes aren't required as the access occur under exclusive but they help set the principle in the code base that we privatize extra before using it.

I **like** the principle being set that we always privatize use of `extra` so I'd prefer to keep these changes.

Reply:
Ah I see; got confused because the fact about exclusive vmaccess was explicitly noted.